### PR TITLE
Configure Dependabot to update GitHub Actions and Submodules

### PR DESCRIPTION
**Pull request by sgiehl**
### Description:

This PR configures Dependabot to automatically set up pull requests for updating used GitHub Actions and all Submodules.

For GitHub Actions Dependabot would automatically set up PRs monthly - one PR for each (external) action that can be updated

For Submodules Dependabot would automatically set up on PR weekly, including all available updates for submodules.

The adjustments in submodule configuration was necessary, as dependabot will always update the submodule based on this definition. So setting the master branch there, will cause dependabot updating the submodule to that branch.

Note: Using Dependabot to update composer and npm packages will be added at a later point.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by michalkleiner** [comment]
This looks ok to me, do you have a link to a PR within your fork to see how the commit messages and PR titles look like?

**Comment by sgiehl** [comment]
@michalkleiner A submodule PR will look like https://github.com/sgiehl/matomo/pull/47
And action update PRs like https://github.com/sgiehl/matomo/pull/10 or https://github.com/sgiehl/matomo/pull/9

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+version: 2
+updates:       
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "monthly"
+    commit-message:
+      prefix: "[GitHub Actions] "
```
Comment (michalkleiner) [inline .github/dependabot.yml:8]: This prefix is not visible in the fork PRs, has it been added after the PRs were created or it doesn't work perhaps?

Comment (sgiehl) [reply .github/dependabot.yml:8]: Guess I added it afterwards. I'll let the PR be recreated

Comment (sgiehl) [reply .github/dependabot.yml:8]: Here is one created later: https://github.com/sgiehl/matomo/pull/14

**Review by michalkleiner — APPROVED** [review]
👍

**Pull request merged**